### ISSUE TITLE
Type error when using spread operator of correct length

### ISSUE THREAD

**Issue by esromneb** [microsoft/TypeScript#38195]
I am trying to do a simple spread operator but I am getting typescript errors.

**TypeScript Version:**  3.8.3

**Code**
```typescript
type Vec2 = [number,number];
type Vec4 = [number,number,number,number];

const a: Vec2  = [0,1];
const b: Vec2  = [2,3];
const broken: Vec4 = [...a,...b];
const works: Vec4 = [a[0],a[1],b[0],b[1]];
```

```ts
// A *self-contained* demonstration of the problem follows...
// Test this by running `tsc` on the command-line, rather than through another build tool such as Gulp, Webpack, etc.
```

**Expected behavior:**
I expect this to work.  I expect broken and works to both have the value of [0,1,2,3]

**Actual behavior:**
The broken line gives me this error:
```
[tsl] ERROR in /mnt/src/main.ts(6,7)
      TS2739: Type 'number[]' is missing the following properties from type '[number, number, number, number]': 0, 1, 2, 3

```

**Playground Link:** 

[Playground Link](https://www.typescriptlang.org/play/?ssl=1&ssc=1&pln=36&pc=1#)

**Related Issues:**

**Comment by jcalz** [comment]
The new issue template asks for search terms and related issues in order to help the team (which I'm not part of, just a nosy bystander here 🙋‍♂️👋) triage and respond to them.   

This is likely a duplicate of #27859; tuples don't currently preserve their tuple-ness when spread.

**Comment by RyanCavanaugh** [comment]
This has been fixed; you can select "Nightly" in the version selector on the Playground to see that the error no longer occurs. Thanks!

**Comment by jcalz** [comment]
Fixed in #36861 ? Or somewhere else?
